### PR TITLE
nix build 

### PR DESCRIPTION
**Pull request by AleXoundOS**
Add default.nix required for build with nix package manager. Thus, no other prerequisites required (like compiler or external libraries) for the build.

### CODE REVIEW

**Review by raboof — APPROVED** [review]
I like nix (I use NixOS full-time), but I'm not up-to-date on the conventions for keeping a default.nix in a project repo, so I'll just trust you on this :). Thanks!

**Comment by raboof** [comment]
(and thanks for fixing the file permissions)

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    path: type: baseNameOf path != ".git" && type != "symlink"
+  ) ./.;
+
+  buildInputs = [ pkgs.zlib ];
+
+  installFlags = [ "PREFIX=$(out)" ];
+
+  meta = {
+    description = "Library for decompressing sfArk soundfonts";
+    platforms = pkgs.lib.platforms.linux;
```
Comment (raboof) [inline default.nix:22]: Maybe add license and homepage fields?

Comment (AleXoundOS) [reply default.nix:22]: Yeah, these would be required for adding a package to https://github.com/NixOS/nixpkgs/. Later I will write proper nixpkgs packages for both sfArkLib and sfArkXTm for pull request into nixpkgs if you let.

Here, `default.nix` is for the purpose of local testing/usage, so the `meta` attributes are not used in any way indeed.

**Comment by AleXoundOS** [comment]
Ah, I see you're quite an experienced nix user ;)

As for conventions, I'm not a representative to tell the truth but it is better to have default.nix / shell.nix than not.
I just followed [the sfArk decompression guide](https://blogs.fsfe.org/samtuke/?p=589), wrote nix expressions and decided to share them for others who encounter the same task. Still, it would be better to add proper packages into nixpkgs. Though, I haven't done this yet in public nixpkgs, I have experience doing this in a company. @raboof, if you want, you can be the maintainer of the nethogs, notion, sfArkLib and sfArkXTm nixpkgs packages, since you can definitely manage this ^^.

**Comment by raboof** [comment]
> I'm not a representative to tell the truth but it is better to have default.nix / shell.nix than not.

Sure!

> I just followed [the sfArk decompression guide](https://blogs.fsfe.org/samtuke/?p=589)

Neat, I didn't know about that.

> wrote nix expressions and decided to share them for others who encounter the same task. Still, it would be better to add proper packages into nixpkgs. Though, I haven't done this yet in public nixpkgs, I have experience doing this in a company. @raboof, if you want, you can be the maintainer of the nethogs, notion, sfArkLib and sfArkXTm nixpkgs packages, since you can definitely manage this ^^.

I'd be happy to review & co-maintain if you want to add it!

**Pull request merged**